torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -48,12 +48,20 @@ struct AutoNcclGroup {
   }
 };
 
+#if defined(NCCL_MAJOR) && ((NCCL_MAJOR > 2) || \
+                            (NCCL_MAJOR == 2) && (NCCL_MINOR >= 10))
+#define NCCL_HAS_AVG 1
```

Comment:
Hmm, what would be the case when ProcessGroupNCCL is getting compiled, but `NCCL_MAJOR` is not defined?

Reply:
ProcessGroupNCCL makes a lot of raw nccl library calls, so I don't think it can compile without including the nccl library's <nccl.h>.
I think <nccl.h> enters ProcessGroupNCCL.cpp by way of ProcessGroupNCCL.hpp which includes NCCLUtils.hpp which includes <nccl.h>.
My current understanding is, NCCL_MAJOR and MINOR are also defined by <nccl.h>, so ProcessGroupNCCL.cpp can't compile at all without NCCL_MAJOR and MINOR being present.

Or are NCCL_MAJOR and MINOR (re)defined by Pytorch headers somewhere else?